lib/Dialect/ESI/ESITypes.cpp

```diff
@@ -164,6 +164,17 @@ inline ::llvm::raw_ostream &operator<<(::llvm::raw_ostream &p,
 }
 } // namespace llvm
 
+ChannelBundleType ChannelBundleType::getReversed() const {
+  SmallVector<BundledChannel, 4> reversed;
+  for (auto channel : getChannels()) {
+    ChannelDirection rev = channel.direction == ChannelDirection::from
+                               ? ChannelDirection::to
+                               : ChannelDirection::from;
```

Comment:
mega-nit: extract this to a `static ChannelDirection flip(ChannelDirection)` function.